### PR TITLE
Add MTU field to route type definition

### PR DESCRIPTION
**Pull request by Joffref**
Fixes: #1004

### CODE REVIEW

File: pkg/types/020/types_test.go [hunk 1]
```diff
@@ -79,7 +79,8 @@ func testResult(resultCNIVersion, jsonCNIVersion string) (*types020.Result, stri
         "routes": [
             {
                 "dst": "15.5.6.0/24",
-                "gw": "15.5.6.8"
+                "gw": "15.5.6.8",
+				"mtu": 1024
```
Comment (aojea) [inline pkg/types/020/types_test.go:83]: indentation looks wrong

File: pkg/version/legacy_examples/examples.go [hunk 2]
```diff
@@ -135,7 +135,8 @@ var ExpectedResult = &types020.Result{
 					IP:   net.ParseIP("0.0.0.0"),
 					Mask: net.CIDRMask(0, 32),
 				},
-				GW: net.ParseIP("10.1.0.1"),
+				GW:  net.ParseIP("10.1.0.1"),
+				MTU: 0,
```
Comment (aojea) [inline pkg/version/legacy_examples/examples.go:139]: does this make sense? to pass 0?

**Comment by aojea** [comment]
have you tested this? it does not look like https://github.com/containernetworking/cni/pull/831

It also required to update Spec.md

/cc @squeed @dcbw

File: pkg/types/020/types_test.go [hunk 3]
```diff
@@ -53,14 +53,14 @@ func testResult(resultCNIVersion, jsonCNIVersion string) (*types020.Result, stri
 			IP:      *ipv4,
 			Gateway: net.ParseIP("1.2.3.1"),
 			Routes: []types.Route{
-				{Dst: *routev4, GW: routegwv4},
+				{Dst: *routev4, GW: routegwv4, MTU: 1024},
```
Comment (squeed) [inline pkg/types/020/types_test.go:56]: We should probably not add any new fields to v0.2.0 and v0.4.0 -- what do you think, @dcbw ?

**Comment by squeed** [comment]
This looks basically good. We also need to describe this in SPEC.md. Shouldn't be more than a line or two needed.

File: pkg/types/020/types_test.go (hunk 3 above)
Comment (squeed) [reply pkg/types/020/types_test.go:56]: @Joffref yes please.
(Sorry, I somehow missed your question!)

Comment (Joffref) [reply pkg/types/020/types_test.go]: Change made!

File: pkg/types/020/types_test.go [hunk 4]
```diff
@@ -17,13 +17,13 @@ package types020_test
 import (
 	"encoding/json"
 	"fmt"
+	types020 "github.com/containernetworking/cni/pkg/types/020"
```
Comment (aojea) [inline pkg/types/020/types_test.go:20]: nit this change same unrelated

**Comment by Joffref** [comment]
Added in #1041

**Pull request closed without merging**